lib/widgets/autocomplete.dart

```diff
@@ -1,43 +1,139 @@
 import 'package:flutter/material.dart';
 
+import '../api/model/model.dart';
+import '../api/route/streams.dart';
 import 'content.dart';
 import 'store.dart';
 import '../model/autocomplete.dart';
 import '../model/compose.dart';
 import '../model/narrow.dart';
 import 'compose_box.dart';
 
-class ComposeAutocomplete extends StatefulWidget {
-  const ComposeAutocomplete({
+class ComposeAutocomplete extends AutocompleteField<MentionAutocompleteQuery, MentionAutocompleteResult, User> {
+  ComposeAutocomplete({
+    super.key,
+    required Narrow narrow,
+    required ComposeContentController controller,
+    required super.focusNode,
+    required super.fieldViewBuilder
+  }) : super(
+    controller: controller,
+    getAutocompleteIntent: () => controller.autocompleteIntent(),
+    viewModelBuilder: (context) {
+      final store = PerAccountStoreWidget.of(context);
+      return MentionAutocompleteView.init(store: store, narrow: narrow);
+    },
+    itemBuilder: (context, index, option) {
+      Widget avatar;
+      String label;
+      switch (option) {
+        case UserMentionAutocompleteResult(:var userId):
+          avatar = Avatar(userId: userId, size: 32, borderRadius: 3);
```

Comment:
This gets fairly messy to read, as a long function expression in a constructor's initializer list.

Let's define `itemBuilder` and friends on the base class as methods instead of fields, so that the subclasses can just override those methods. See our `MessageActionSheetMenuItemButton` for an example.